newapi/tests/conftest.py

```diff
@@ -53,7 +60,121 @@ def client(app):
 def pytest_addoption(parser):
     parser.addoption("--ghpr", action="store_true", help="enable GitHub integ tests")
     parser.addoption("--proddb", action="store_true", help="uses data from prod DB")
+    parser.addoption("--create-db", action="store_true", help="populate a dedicated DB")
+
 
 def pytest_configure(config):
     pytest.run_ghpr = config.getoption("--ghpr")
     pytest.proddb = config.getoption("--proddb")
+    assert pytest.proddb is False, "--proddb is disabled"
+    pytest.create_db = config.getoption("--create-db")
+
+
+def sudopg(cmd, check=True):
+    cmd = ["/usr/bin/sudo", "-u", "postgres", "psql", "-c", cmd]
+    print(cmd)
+    out = subprocess.run(cmd, check=check, stdout=PIPE, stderr=PIPE).stdout
+    out = out.decode().strip()
+    if out:
+        print(out)
+
+
+@pytest.fixture(scope="session")
+def setup_database_part_1():
+    # Create database and users.
+    # Drop and recreate database if exists.
+    if not pytest.create_db:
+        return
+
+    if os.path.exists("/usr/bin/sudo"):
+        print("Creating PostgreSQL user and database")
+        sudopg("DROP DATABASE IF EXISTS oonitestdb", check=True)
```

Comment:
I suspect these will not work because you are running them from inside of the ooni_api container instead of the DB container. To target the `db` container you will have to use network based authentication rather than socket based.

Reply:
"sudo" is being used or not based on the environment in order to be able to run the tests in different environments. See the comment " On github sudo is missing and the database is already created "